overlord/ifacestate/ifacemgr.go

```diff
@@ -137,6 +141,20 @@ func Manager(s *state.State, hookManager *hookstate.HookManager, runner *state.T
 	return m, nil
 }
 
+// AppArmorPromptingRunning returns true if prompting is running.
+func (m *InterfaceManager) AppArmorPromptingRunning() bool {
+	return useAppArmorPrompting(m)
```

Comment:
this is a bit of a weird one because the first time this is called we need the state locked and after that not strictly, my preference would be to expose only the cached value instead maybe?

Reply:
The problem is how to ensure that the cached value has actually been computed. If we're fairly sure that the call to `useAppArmorPrompting` in `StartUp` will be the first time it can be called, then I'm comfortable having this method just export the value, since we don't call this method from within this package.

In fact, we don't call this method from anywhere yet, it is only used from the API. So perhaps just exporting the value is perfectly safe.

We could do something safer by adding a channel which is closed when the `sync.Once` in `useAppArmorPrompting` sets the value, but then there's a chance of blocking during the `AppArmorPromptingRunning` call if for some reason it happens before `StartUp`.

Another alternative would be to do something like this:
```go
func (m *InterfaceManager) AppArmorPromptingRunning() bool {
	m.useAppArmorPromptingChecker.Do(func() {
		m.state.Lock()
		defer m.state.Unlock()
		useAppArmorPrompting(m)
	}
	return m.useAppArmorPromptingValue
}
```

So I see a few options:
1. Just return the current value, even if it's uninitialized
2. Add a channel which `AppArmorPromptingRunning` waits to close before returning the current value
3. Make `AppArmorPromptingRunning` call a closure which locks state and then calls `m.useAppArmorPrompting`

Reply:
AFAIU first time this will be called in StartUp() of ifacemgr. I don't think anything else may need it.